### ISSUE TITLE
krun "just a string, without any options"

### ISSUE THREAD

**Issue by grosu** [kframework/k-legacy#196]
I would like to propose a very simple but useful enhancement of krun.  Basically, if you pass a string instead of a file as an argument, to automatically assume the parser "kast -e".  That is, instead of asking our users to write

```
$ krun --parser "kast -e" "7 + 3"
```

to only ask them to write

```
$ krun "7 - 3"
```

This is a very low-hanging fruit, which will increase the usability of the K tool.

I found that most of my students in cs422 do not even know the "kast -e" trick, so each time they want to test something simple they still create a file and put the term to rewrite there.

If nobody has anything against this simple enhancement, I think we should go for it.

**Comment by adamsmd** [comment]
I agree with this idea.  (When I saw you write `krun --parser "kast -e" 
"7 + 3"`, I thought about proposing this idea.)  However, I might 
suggest making it `krun -e "7 - 3"` so there is no chance of confusion 
between a filename and an expression on the command line.

How much of the rest of "kast" could also be merged cleanly into krun?

**Comment by dlucanu** [comment]
I used only very rare this option, but I agree that such functionality 
is useful.
I wished to uses in classes, but a command like  krun --parser "kast -e" 
"7 + 3" is very confusing for the beginners.
I guess that command of the form

kast -cPGM "7+3"

is more uniform with the other options.

Dorel

**Comment by traiansf** [comment]
I also believe it should be prefixed by an option.

If a file name has spaces (or other strange characters) in it, esp. on
windows, the default way of passing it to krun would be to quote it.  That
would work in linux, too, although there one sometimes escape the spaces.

2013/12/4 Dorel Lucanu notifications@github.com

**Comment by grosu** [comment]
OK, the ideal here would then be to allow both Micheal's and Dorel's suggestions.  Actually, Dorel's already works, right?  But for Michael's, we need to make sure that kast's options are not overridden by krun's options, and thus can be simply inherited by krun.  Then any option that the user gives to krun is passed to ktest if it is a kstest option.  Then krun -e "7 + 3" will work nicely, because -e is ktest option, so it will be passed to ktest.

How difficult would it be to achieve that?

Grigore
